subsys/bluetooth/mesh/cfg_srv.c

```diff
@@ -1412,49 +1414,63 @@ static int mod_sub_va_add(struct bt_mesh_model *model,
 	if (!elem) {
 		mod = NULL;
 		vnd = (buf->len == 4U);
-		sub_addr = BT_MESH_ADDR_UNASSIGNED;
 		status = STATUS_INVALID_ADDRESS;
 		goto send_status;
 	}
 
 	mod = get_model(elem, buf, &vnd);
 	if (!mod) {
-		sub_addr = BT_MESH_ADDR_UNASSIGNED;
 		status = STATUS_INVALID_MODEL;
 		goto send_status;
 	}
 
-	status = bt_mesh_va_add(label_uuid, &sub_addr);
+	status = bt_mesh_va_add(uuid, &va);
 	if (status != STATUS_SUCCESS) {
 		goto send_status;
 	}
 
-	if (bt_mesh_model_find_group(&mod, sub_addr)) {
+	if (bt_mesh_model_find_uuid(&mod, va->uuid)) {
 		/* Tried to add existing subscription */
 		status = STATUS_SUCCESS;
-		bt_mesh_va_del(label_uuid, NULL);
+		(void)bt_mesh_va_del(va->uuid);
 		goto send_status;
 	}
 
-
-	entry = bt_mesh_model_find_group(&mod, BT_MESH_ADDR_UNASSIGNED);
-	if (!entry) {
+	label_entry = bt_mesh_model_find_uuid(&mod, NULL);
+	if (!label_entry) {
 		status = STATUS_INSUFF_RESOURCES;
-		bt_mesh_va_del(label_uuid, NULL);
+		(void)bt_mesh_va_del(va->uuid);
 		goto send_status;
 	}
 
-	*entry = sub_addr;
+	group_entry = NULL;
 
-	if (IS_ENABLED(CONFIG_BT_MESH_LOW_POWER)) {
-		bt_mesh_lpn_group_add(sub_addr);
+	for (int i = 0; i < mod->groups_cnt; i++) {
+		if (mod->groups[i] == BT_MESH_ADDR_UNASSIGNED) {
+			group_entry = &mod->groups[i];
+			break;
+		}
+	}
+
+	/* bt_mesh_model_find_uuid() should find a model where both, uuids and groups lists have
+	 * empty entry.
+	 */
+	__ASSERT_NO_MSG(group_entry);
```

Comment:
The device shouldn't fall into assertion. It should return the error status back.